python/RESOURCE_MANAGED_BY_CLOUDFORMATION/RESOURCE_MANAGED_BY_CLOUDFORMATION.py

```diff
@@ -0,0 +1,495 @@
+            roleList.append(role)
+    return roleList
+
+
+def build_iam_managed_policy_list():
+    policyList = []
+    iam = boto3.client('iam')
+    # print(iam.list_policies()['IsTruncated'])\
+    policyListResult = iam.list_policies(
+        Scope='Local',
+    )
+    for policy in policyListResult['Policies']:
+        policyList.append(policy)
+    return policyList
+
+
+def check_resource_managaed_by_cloudformation(resourceType, resourceList, resourceName, resourceId):
+    compliance_result_for_type = []
+    cfnResources = allResources.get(resourceType, 'not found')
+    if cfnResources != 'not found':
+        for resource in resourceList:
+            if (resource[resourceName] in cfnResources):
+                compliance_result_for_type.append({
+                    'ComplianceResourceType': resourceType,
+                    'ComplianceResourceId': resource[resourceId],
+                    'ComplianceType': 'COMPLIANT',
+                    'OrderingTimestamp': str(datetime.datetime.now()),
+                })
+            else:
+                compliance_result_for_type.append({
+                    'ComplianceResourceType': resourceType,
+                    'ComplianceResourceId': resource[resourceId],
+                    'ComplianceType': 'NON_COMPLIANT',
+                    'OrderingTimestamp': str(datetime.datetime.now()),
+                })
+    return compliance_result_for_type
+
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+
+    compliance_result = []
+    build_cloudformation_resource_list()
```

Comment:
Avoid to use a global variable. Can you pass and retrieve actual variables?

Reply:
I was not sure if I would need to address result pagination with the CloudFormation APIs. In my tests I was able to return many resources from many stacks without pagination. For very large CloudFormation deployments, there may be a limit to how many resources can be returned in a single API call. For now I've removed this block.

Reply:
Resolved in latest commit